R/fda-table_20.R

```diff
@@ -0,0 +1,136 @@
+# Pre-processing
+
+library(dplyr)
+library(scda)
+library(tern)
+
+adsl <- scda::synthetic_cdisc_dataset("rcd_2022_10_13", "adsl") %>%
+  df_explicit_na()
+
+adae <- scda::synthetic_cdisc_dataset("rcd_2022_10_13", "adae") %>%
+  mutate(
+    aesev = factor(
+      case_when(
+        AESEV == "MILD" ~ "Mild",
+        AESEV == "MODERATE" ~ "Moderate",
+        AESEV == "SEVERE" ~ "Severe"
+      ),
+      levels = c("Severe", "Moderate", "Mild")
+    ),
+    # create adverse events of special interest flag
+    AESIFL = case_when(
+      AESOC == "cl A" ~ "Y",
+      AESOC == "cl D" ~ "Y",
+      TRUE ~ "N"
+    ),
+    # create flag for laboratory assessment
+    AELABFL = sample(c("Y", "N"), n(), replace = TRUE)
+  ) %>%
+  filter(
+    SAFFL == "Y",
+    AESIFL == "Y"
+  ) %>%
+  df_explicit_na()
+
+
+lyt <- basic_table(show_colcounts = TRUE) %>%
+  split_cols_by(var = "ARM") %>%
+  append_topleft(c("", "AESI Assessment")) %>%
+  count_patients_with_event(
+    "USUBJID",
+    filters = c("AESIFL" = "Y"),
+    .labels = "AE grouping related to AESI",
+    denom = "N_col",
+    table_names = "tbl_aesi"
+  ) %>%
+  count_occurrences(
+    "AEDECOD",
+    .indent_mods = 1L
+  ) %>%
+  count_occurrences(
```

Comment:
`count_occurrences_by_grade` can be used here instead of `count_occurrences` so that only the highest severity grade is counted for each patient and only one grade is kept for each patient.

```suggestion
  count_occurrences_by_grade(
```